### PR TITLE
Legends map options

### LINKED ISSUE #9751: Add legends options to map options
After #9584, we need to add the option in order to customise the public view.

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #9751 #9223.

legends-map-options

It changes a bit the way we save the layer selector option. Because it's coupled with the legend block, it doesn't make sense to handle it different. Remember that overlays create views with order and positions. Until now, the layer selector was an overlay in cartodb.js, and but not legends. This PR treats them both as they would be outside overlays. In the builder this means we are saving both options in the map definition model. So the `vizjson` need to reflect this change and it should add the proper `layer_selector`.

~~If we decide this is the way to go, the changes in the vizjson need to be done in order to persist the user's option.~~

Also, as part of this PR, I fixed how we initialise the custom legend model, in the same manner of the rest of models. This avoid an empty categories in the form when the custom option is rendered.

This PR depends on https://github.com/CartoDB/cartodb/pull/9926 and viceversa. Also depends on https://github.com/CartoDB/cartodb.js/pull/1425

cc @xavijam

### CODE REVIEW

**Comment by xavijam** [comment]
I'd remove the "layer selector" option because as you have pointed, it does not make any sense to treat them in two different ways if they are in the same component.

**Comment by nobuti** [comment]
This is ready for another pass.

File: lib/assets/javascripts/cartodb3/editor.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ UserGroupFetcher.track({
 // from the given vizJSON, but enabling legends behind a feature flag
 vizJSON = _.extend(
   vizJSON, {
-    legends: userModel.featureEnabled('builder-legends')
+    legends: userModel.featureEnabled('builder-legends') && vizJSON.legends
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor.js]: Let's remove feature flag please :)

File: lib/assets/javascripts/cartodb3/public_editor.js [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
-  {
-    legends: false
-  }
-);
+var vizJSON = window.vizJSON;
 var dashboardOpts = {
   no_cdn: false,
   cartodb_logo: false,
-  renderMenu: true,
+  renderMenu: vizJSON.embed_options.dashboard_menu,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/public_editor.js]: cc @saleiva

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/public_editor.js]: And enable legends if they are not.

**Comment by nobuti** [comment]
Getting the do not merge label back again. This PR depends on two other PR, so we have to wait for them.

**Comment by nobuti** [comment]
Current behaviour after fixing it:

map-legends-options

**Comment by nobuti** [comment]
Frontend tests passing 🚀

**Pull request merged**